packages/playwright/src/isomorphic/teleReceiver.ts

```diff
@@ -512,8 +516,11 @@ class TeleTestStep implements reporterTypes.TestStep {
   parent: reporterTypes.TestStep | undefined;
   duration: number = -1;
   steps: reporterTypes.TestStep[] = [];
+  attachments: reporterTypes.TestStep['attachments'] = [];
```

Comment:
Let's make this a getter and compute lazily upon access? There is a memory pressure problem when merging a lot of large reports, and this class takes the most memory because there are a lot of steps.

Reply:
I made it a getter in https://github.com/microsoft/playwright/pull/34037/commits/365b30f64ffb9c113b92c337d8b5d54c80096c99. We can look into refactoring the rest of this as well, so that `TeleTestStep` only keeps `_startPayload` and `_endPayload` references.